File: VB/WebSite/App_Code/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

''' <summary>
''' Summary description for XtraReport1
''' </summary>
Public Class XtraReport1
	Inherits DevExpress.XtraReports.UI.XtraReport
	Private Detail As DevExpress.XtraReports.UI.DetailBand
	Private TopMargin As DevExpress.XtraReports.UI.TopMarginBand
	Private BottomMargin As DevExpress.XtraReports.UI.BottomMarginBand
	Private northwind1 As Northwind
	Private xrPictureBox1 As XRPictureBox
	Private xrLabel2 As XRLabel
	Private xrLabel1 As XRLabel
	Private categoriesTableAdapter1 As NorthwindTableAdapters.CategoriesTableAdapter
	Private DetailReport As DetailReportBand
	Private Detail1 As DetailBand
	Private xrLabel3 As XRLabel
	Private xrLabel4 As XRLabel
	Private productsTableAdapter1 As NorthwindTableAdapters.ProductsTableAdapter
	Private CategoryId As DevExpress.XtraReports.Parameters.Parameter
	''' <summary>
	''' Required designer variable.
	''' </summary>
	Private components As System.ComponentModel.IContainer = Nothing

	Public Sub New()
		InitializeComponent()
		'
		' TODO: Add constructor logic here
		'
	End Sub

	''' <summary> 
	''' Clean up any resources being used.
	''' </summary>
	''' <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
	Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
		If disposing AndAlso (components IsNot Nothing) Then
			components.Dispose()
		End If
		MyBase.Dispose(disposing)
	End Sub

	#Region "Designer generated code"

	''' <summary>
	''' Required method for Designer support - do not modify
	''' the contents of this method with the code editor.
	''' </summary>
	Private Sub InitializeComponent()
		Dim resourceFileName As String = "XtraReport1.resx"
		Me.Detail = New DevExpress.XtraReports.UI.DetailBand()
		Me.xrPictureBox1 = New DevExpress.XtraReports.UI.XRPictureBox()
		Me.xrLabel2 = New DevExpress.XtraReports.UI.XRLabel()
		Me.xrLabel1 = New DevExpress.XtraReports.UI.XRLabel()
		Me.TopMargin = New DevExpress.XtraReports.UI.TopMarginBand()
		Me.BottomMargin = New DevExpress.XtraReports.UI.BottomMarginBand()
		Me.northwind1 = New Northwind()
		Me.categoriesTableAdapter1 = New NorthwindTableAdapters.CategoriesTableAdapter()
		Me.DetailReport = New DevExpress.XtraReports.UI.DetailReportBand()
		Me.Detail1 = New DevExpress.XtraReports.UI.DetailBand()
		Me.xrLabel4 = New DevExpress.XtraReports.UI.XRLabel()
		Me.xrLabel3 = New DevExpress.XtraReports.UI.XRLabel()
		Me.productsTableAdapter1 = New NorthwindTableAdapters.ProductsTableAdapter()
		Me.CategoryId = New DevExpress.XtraReports.Parameters.Parameter()
		CType(Me.northwind1, System.ComponentModel.ISupportInitialize).BeginInit()
		CType(Me, System.ComponentModel.ISupportInitialize).BeginInit()
		' 
		' Detail
		' 
		Me.Detail.Controls.AddRange(New DevExpress.XtraReports.UI.XRControl() { Me.xrPictureBox1, Me.xrLabel2, Me.xrLabel1})
		Me.Detail.HeightF = 91F
		Me.Detail.Name = "Detail"
		Me.Detail.Padding = New DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F)
		Me.Detail.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft
		' 
		' xrPictureBox1
		' 
		Me.xrPictureBox1.DataBindings.AddRange(New DevExpress.XtraReports.UI.XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Image", Nothing, "Categories.Picture")})
		Me.xrPictureBox1.LocationFloat = New DevExpress.Utils.PointFloat(444.1667F, 10.00001F)
		Me.xrPictureBox1.Name = "xrPictureBox1"
		Me.xrPictureBox1.SizeF = New System.Drawing.SizeF(195.8333F, 63.99998F)
		' 
		' xrLabel2
		' 
		Me.xrLabel2.DataBindings.AddRange(New DevExpress.XtraReports.UI.XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Text", Nothing, "Categories.Description")})
		Me.xrLabel2.LocationFloat = New DevExpress.Utils.PointFloat(10.00001F, 50.99999F)
		Me.xrLabel2.Name = "xrLabel2"
		Me.xrLabel2.Padding = New DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F)
		Me.xrLabel2.SizeF = New System.Drawing.SizeF(365.625F, 23F)
		Me.xrLabel2.Text = "xrLabel2"
		' 
		' xrLabel1
		' 
		Me.xrLabel1.DataBindings.AddRange(New DevExpress.XtraReports.UI.XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Text", Nothing, "Categories.CategoryName")})
		Me.xrLabel1.Font = New System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, (CByte(204)))
		Me.xrLabel1.ForeColor = System.Drawing.Color.Blue
		Me.xrLabel1.LocationFloat = New DevExpress.Utils.PointFloat(10.00001F, 10.00001F)
		Me.xrLabel1.Name = "xrLabel1"
		Me.xrLabel1.Padding = New DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F)
		Me.xrLabel1.SizeF = New System.Drawing.SizeF(209.375F, 23F)
		Me.xrLabel1.StylePriority.UseFont = False
		Me.xrLabel1.StylePriority.UseForeColor = False
		Me.xrLabel1.Text = "xrLabel1"
		' 
		' TopMargin
		' 
		Me.TopMargin.Name = "TopMargin"
		Me.TopMargin.Padding = New DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F)
		Me.TopMargin.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft
		' 
		' BottomMargin
		' 
		Me.BottomMargin.Name = "BottomMargin"
		Me.BottomMargin.Padding = New DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F)
		Me.BottomMargin.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft
		' 
		' northwind1
		' 
		Me.northwind1.DataSetName = "Northwind"
		Me.northwind1.SchemaSerializationMode = System.Data.SchemaSerializationMode.IncludeSchema
		' 
		' categoriesTableAdapter1
		' 
		Me.categoriesTableAdapter1.ClearBeforeFill = True
		' 
		' DetailReport
		' 
		Me.DetailReport.Bands.AddRange(New DevExpress.XtraReports.UI.Band() { Me.Detail1})
		Me.DetailReport.DataAdapter = Me.productsTableAdapter1
		Me.DetailReport.DataMember = "Categories.CategoriesProducts"
		Me.DetailReport.DataSource = Me.northwind1
		Me.DetailReport.Level = 0
		Me.DetailReport.Name = "DetailReport"
		' 
		' Detail1
		' 
		Me.Detail1.Controls.AddRange(New DevExpress.XtraReports.UI.XRControl() { Me.xrLabel4, Me.xrLabel3})
		Me.Detail1.HeightF = 23F
		Me.Detail1.Name = "Detail1"
		' 
		' xrLabel4
		' 
		Me.xrLabel4.DataBindings.AddRange(New DevExpress.XtraReports.UI.XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Text", Nothing, "Categories.CategoriesProducts.UnitPrice")})
		Me.xrLabel4.LocationFloat = New DevExpress.Utils.PointFloat(444.1667F, 0F)
		Me.xrLabel4.Name = "xrLabel4"
		Me.xrLabel4.Padding = New DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F)
		Me.xrLabel4.SizeF = New System.Drawing.SizeF(100F, 23F)
		Me.xrLabel4.Text = "xrLabel4"
		' 
		' xrLabel3
		' 
		Me.xrLabel3.DataBindings.AddRange(New DevExpress.XtraReports.UI.XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Text", Nothing, "Categories.CategoriesProducts.ProductName")})
		Me.xrLabel3.LocationFloat = New DevExpress.Utils.PointFloat(10.00001F, 0F)
		Me.xrLabel3.Name = "xrLabel3"
		Me.xrLabel3.Padding = New DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F)
		Me.xrLabel3.SizeF = New System.Drawing.SizeF(279.1667F, 23F)
		Me.xrLabel3.Text = "xrLabel3"
		' 
		' productsTableAdapter1
		' 
		Me.productsTableAdapter1.ClearBeforeFill = True
		' 
		' CategoryId
		' 
		Me.CategoryId.Name = "CategoryId"
		Me.CategoryId.ParameterType = DevExpress.XtraReports.Parameters.ParameterType.Int32
		Me.CategoryId.Value = 1
		' 
		' XtraReport1
		' 
		Me.Bands.AddRange(New DevExpress.XtraReports.UI.Band() { Me.Detail, Me.TopMargin, Me.BottomMargin, Me.DetailReport})
		Me.DataAdapter = Me.categoriesTableAdapter1
		Me.DataMember = "Categories"
		Me.DataSource = Me.northwind1
		Me.FilterString = "[CategoryID] = ?CategoryId"
		Me.Parameters.AddRange(New DevExpress.XtraReports.Parameters.Parameter() { Me.CategoryId})
		Me.Version = "10.1"
		CType(Me.northwind1, System.ComponentModel.ISupportInitialize).EndInit()
		CType(Me, System.ComponentModel.ISupportInitialize).EndInit()

	End Sub

	#End Region
End Class

File: CS/WebSite/App_Code/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

/// <summary>
/// Summary description for XtraReport1
/// </summary>
public class XtraReport1 : DevExpress.XtraReports.UI.XtraReport
{
	private DevExpress.XtraReports.UI.DetailBand Detail;
	private DevExpress.XtraReports.UI.TopMarginBand TopMargin;
	private DevExpress.XtraReports.UI.BottomMarginBand BottomMargin;
    private Northwind northwind1;
    private XRPictureBox xrPictureBox1;
    private XRLabel xrLabel2;
    private XRLabel xrLabel1;
    private NorthwindTableAdapters.CategoriesTableAdapter categoriesTableAdapter1;
    private DetailReportBand DetailReport;
    private DetailBand Detail1;
    private XRLabel xrLabel3;
    private XRLabel xrLabel4;
    private NorthwindTableAdapters.ProductsTableAdapter productsTableAdapter1;
    private DevExpress.XtraReports.Parameters.Parameter CategoryId;
	/// <summary>
	/// Required designer variable.
	/// </summary>
	private System.ComponentModel.IContainer components = null;

	public XtraReport1()
	{
		InitializeComponent();
		//
		// TODO: Add constructor logic here
		//
	}
	
	/// <summary> 
	/// Clean up any resources being used.
	/// </summary>
	/// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
	protected override void Dispose(bool disposing) {
		if (disposing && (components != null)) {
			components.Dispose();
		}
		base.Dispose(disposing);
	}

	#region Designer generated code

	/// <summary>
	/// Required method for Designer support - do not modify
	/// the contents of this method with the code editor.
	/// </summary>
	private void InitializeComponent() {
        string resourceFileName = "XtraReport1.resx";
        this.Detail = new DevExpress.XtraReports.UI.DetailBand();
        this.xrPictureBox1 = new DevExpress.XtraReports.UI.XRPictureBox();
        this.xrLabel2 = new DevExpress.XtraReports.UI.XRLabel();
        this.xrLabel1 = new DevExpress.XtraReports.UI.XRLabel();
        this.TopMargin = new DevExpress.XtraReports.UI.TopMarginBand();
        this.BottomMargin = new DevExpress.XtraReports.UI.BottomMarginBand();
        this.northwind1 = new Northwind();
        this.categoriesTableAdapter1 = new NorthwindTableAdapters.CategoriesTableAdapter();
        this.DetailReport = new DevExpress.XtraReports.UI.DetailReportBand();
        this.Detail1 = new DevExpress.XtraReports.UI.DetailBand();
        this.xrLabel4 = new DevExpress.XtraReports.UI.XRLabel();
        this.xrLabel3 = new DevExpress.XtraReports.UI.XRLabel();
        this.productsTableAdapter1 = new NorthwindTableAdapters.ProductsTableAdapter();
        this.CategoryId = new DevExpress.XtraReports.Parameters.Parameter();
        ((System.ComponentModel.ISupportInitialize)(this.northwind1)).BeginInit();
        ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
        // 
        // Detail
        // 
        this.Detail.Controls.AddRange(new DevExpress.XtraReports.UI.XRControl[] {
            this.xrPictureBox1,
            this.xrLabel2,
            this.xrLabel1});
        this.Detail.HeightF = 91F;
        this.Detail.Name = "Detail";
        this.Detail.Padding = new DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F);
        this.Detail.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft;
        // 
        // xrPictureBox1
        // 
        this.xrPictureBox1.DataBindings.AddRange(new DevExpress.XtraReports.UI.XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Image", null, "Categories.Picture")});
        this.xrPictureBox1.LocationFloat = new DevExpress.Utils.PointFloat(444.1667F, 10.00001F);
        this.xrPictureBox1.Name = "xrPictureBox1";
        this.xrPictureBox1.SizeF = new System.Drawing.SizeF(195.8333F, 63.99998F);
        // 
        // xrLabel2
        // 
        this.xrLabel2.DataBindings.AddRange(new DevExpress.XtraReports.UI.XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Text", null, "Categories.Description")});
        this.xrLabel2.LocationFloat = new DevExpress.Utils.PointFloat(10.00001F, 50.99999F);
        this.xrLabel2.Name = "xrLabel2";
        this.xrLabel2.Padding = new DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F);
        this.xrLabel2.SizeF = new System.Drawing.SizeF(365.625F, 23F);
        this.xrLabel2.Text = "xrLabel2";
        // 
        // xrLabel1
        // 
        this.xrLabel1.DataBindings.AddRange(new DevExpress.XtraReports.UI.XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Text", null, "Categories.CategoryName")});
        this.xrLabel1.Font = new System.Drawing.Font("Times New Roman", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(204)));
        this.xrLabel1.ForeColor = System.Drawing.Color.Blue;
        this.xrLabel1.LocationFloat = new DevExpress.Utils.PointFloat(10.00001F, 10.00001F);
        this.xrLabel1.Name = "xrLabel1";
        this.xrLabel1.Padding = new DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F);
        this.xrLabel1.SizeF = new System.Drawing.SizeF(209.375F, 23F);
        this.xrLabel1.StylePriority.UseFont = false;
        this.xrLabel1.StylePriority.UseForeColor = false;
        this.xrLabel1.Text = "xrLabel1";
        // 
        // TopMargin
        // 
        this.TopMargin.Name = "TopMargin";
        this.TopMargin.Padding = new DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F);
        this.TopMargin.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft;
        // 
        // BottomMargin
        // 
        this.BottomMargin.Name = "BottomMargin";
        this.BottomMargin.Padding = new DevExpress.XtraPrinting.PaddingInfo(0, 0, 0, 0, 100F);
        this.BottomMargin.TextAlignment = DevExpress.XtraPrinting.TextAlignment.TopLeft;
        // 
        // northwind1
        // 
        this.northwind1.DataSetName = "Northwind";
        this.northwind1.SchemaSerializationMode = System.Data.SchemaSerializationMode.IncludeSchema;
        // 
        // categoriesTableAdapter1
        // 
        this.categoriesTableAdapter1.ClearBeforeFill = true;
        // 
        // DetailReport
        // 
        this.DetailReport.Bands.AddRange(new DevExpress.XtraReports.UI.Band[] {
            this.Detail1});
        this.DetailReport.DataAdapter = this.productsTableAdapter1;
        this.DetailReport.DataMember = "Categories.CategoriesProducts";
        this.DetailReport.DataSource = this.northwind1;
        this.DetailReport.Level = 0;
        this.DetailReport.Name = "DetailReport";
        // 
        // Detail1
        // 
        this.Detail1.Controls.AddRange(new DevExpress.XtraReports.UI.XRControl[] {
            this.xrLabel4,
            this.xrLabel3});
        this.Detail1.HeightF = 23F;
        this.Detail1.Name = "Detail1";
        // 
        // xrLabel4
        // 
        this.xrLabel4.DataBindings.AddRange(new DevExpress.XtraReports.UI.XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Text", null, "Categories.CategoriesProducts.UnitPrice")});
        this.xrLabel4.LocationFloat = new DevExpress.Utils.PointFloat(444.1667F, 0F);
        this.xrLabel4.Name = "xrLabel4";
        this.xrLabel4.Padding = new DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F);
        this.xrLabel4.SizeF = new System.Drawing.SizeF(100F, 23F);
        this.xrLabel4.Text = "xrLabel4";
        // 
        // xrLabel3
        // 
        this.xrLabel3.DataBindings.AddRange(new DevExpress.XtraReports.UI.XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Text", null, "Categories.CategoriesProducts.ProductName")});
        this.xrLabel3.LocationFloat = new DevExpress.Utils.PointFloat(10.00001F, 0F);
        this.xrLabel3.Name = "xrLabel3";
        this.xrLabel3.Padding = new DevExpress.XtraPrinting.PaddingInfo(2, 2, 0, 0, 100F);
        this.xrLabel3.SizeF = new System.Drawing.SizeF(279.1667F, 23F);
        this.xrLabel3.Text = "xrLabel3";
        // 
        // productsTableAdapter1
        // 
        this.productsTableAdapter1.ClearBeforeFill = true;
        // 
        // CategoryId
        // 
        this.CategoryId.Name = "CategoryId";
        this.CategoryId.ParameterType = DevExpress.XtraReports.Parameters.ParameterType.Int32;
        this.CategoryId.Value = 1;
        // 
        // XtraReport1
        // 
        this.Bands.AddRange(new DevExpress.XtraReports.UI.Band[] {
            this.Detail,
            this.TopMargin,
            this.BottomMargin,
            this.DetailReport});
        this.DataAdapter = this.categoriesTableAdapter1;
        this.DataMember = "Categories";
        this.DataSource = this.northwind1;
        this.FilterString = "[CategoryID] = ?CategoryId";
        this.Parameters.AddRange(new DevExpress.XtraReports.Parameters.Parameter[] {
            this.CategoryId});
        this.Version = "10.1";
        ((System.ComponentModel.ISupportInitialize)(this.northwind1)).EndInit();
        ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

	}

	#endregion
}
